Lib/test/support/__init__.py

```diff
@@ -2873,3 +2874,37 @@ def __iter__(self):
         if self.iter_raises:
             1/0
         return self
+
+
+
+def in_systemd_nspawn_sync_suppressed() -> bool:
+    """
+    Test whether the test suite is runing in systemd-nspawn
+    with ``--suppress-sync=true``.
+
+    This can be used to skip tests that rely on ``fsync()`` calls
+    and similar not being intercepted.
+    """
+
+    if not hasattr(os, "O_SYNC"):
+        return False
+
+    try:
+        with open("/run/systemd/container", "rb") as fp:
+            if fp.read().rstrip() != b"systemd-nspawn":
+                return False
+    except FileNotFoundError:
+        return False
+
+    import errno
+
+    # If systemd-nspawn is used, O_SYNC flag will immediately
+    # trigger EINVAL.  Otherwise, ENOENT will be given instead.
```

Comment:
```suggestion
    # If systemd-nspawn is used, O_SYNC flag will immediately
    # trigger EINVAL.  Otherwise, ENOENT will be given instead.
    import errno
```